### DISCUSSION TITLE
Profissional com github VS profissional sem github

### DISCUSSION THREAD

**Discussion by nandomoreirame** [frontendbr/forum#458] [Discussão]
Eu gostaria de entender melhor alguns pontos e deixar umas dúvidas soltas no ar :D

Gostaria da opinião de vocês...

---

O quanto é importante pra vocês, a nível de contratação, o perfil do github do desenvolvedor?

Será que estamos sendo injustos ou negligentes na hora de comparar um profissional que não tem um profile bom ou uma conta no github com outro que já é mais ativo nas comunidades?

Como vocês avaliam esses profissionais (que não tem um perfil no github) quando são comparados com outros programadores? Será que o fato de ele ser ativo ou ter seus projetos públicos é um ponto forte para a contratação?

Será que estamos deixando de contratar bons profissionais apenas pelo fato de eles não terem um perfil no github ou algo do tipo?

Quando é o momento certo de comparar programadores com currículos distintos (um sem e outro com github)?

E, vocês acham que ser ativo e/ou ter um profile bom no github é um requisito necessário ou desejável para uma vaga?

That's it!

**Comment by ojonathanleal-zz** [comment]
@nandomoreirame ótima pergunta, tenho algo para questionar também que pode aproveitar o gancho.

O que seria um perfil bom no github?

**Comment by nandomoreirame** [comment]
No meu ponto de vista alguém que realmente contribui com a comunidade tem um perfil bom...

Mas na vdd o que gostaria mesmo era entender o quanto isso é crucial para a contratação de um desenvolvedor

**Comment by fdaciuk** [comment]
No meu ponto de vista: isso depende muito da cultura da empresa que está contratando.

Algumas prezam por profissionais que contribuam com open source, porque têm uma cultura de abrir "partes" de seus projetos internos como open source.

Mas contribuir muito não quer dizer a pessoa é um bom profissional, como não ter um perfil ativo no GitHub não significa que o cara não é um bom profissional.

A vantagem de ter um perfil no GitHub, e ter ao menos alguns projetos, é que o seu código fica disponível para quem quiser avaliar. Ao fazer um teste, é possível comparar para ver se o código do GitHub é somente copia-cola ou se o cara desenvolveu mesmo aquilo, e se ele consegue resolver um problema real.

Ter um perfil ativo no GitHub também ajuda que as empresas te achem, principalmente empresas de fora do BR.

Mas na hora do "vamos ver" mesmo de uma contratação, tem muitos mais pontos que pesam na decisão. Ter uma conta no GitHub vai ser apenas a porta de entrada =)

**Comment by matheus-vieira** [comment]
Acho que o GitHub tem mais a relação de quanto aquele perfil se engaja (ou se tem tempo disponível) para compartilhar conhecimento, contribuir em projetos OpenSource etc. Mas mesmo assim não vai ter um relatório completo sobre a pessoa por trás daquele perfil.

Conheço profissionais altamente capacitados que não possuem perfil aqui, mesmo assim contrataria de olho fechado por saber da capacidade. Mas, por outro lado, tem o perfil oposto, pessoas com o perfil mais "ativo", porém não tenho vontade de trabalhar.

Mas gostaria de ouvir mais opniões por aqui tbm!

**Comment by afonsopacifer** [comment]
Outro ponto interessante, é que "ter projetos" no github (ou outro lugar aberto) assim como contribuir para outros projetos legais, te trás várias oportunidades, que vão desde oportunidades de trabalho até de aprender coisas novas. 

É inegável que hoje em dia ter um github ativo trás uma certa visibilidade (o que não garante nada). Não digo que quem tem um github ativo é melhor (acreditem isso não quer dizer nada na real), mas sim que te trás **visibilidade**, algo que alguém q não compartilha código não tem.

Sobre o que seria um **github bom**:

Definitivamente não tem haver com quadradinhos verdes.. longe disso.. tem haver com vc mostrar interesse através da criação de projetos ou de de contribuições. Basicamente se o cara contribui ou criar projetos para solucionar problemas ou para testar novas tecnologias regularmente, demonstra que ele é alguém que no mínimo ama o que faz, pode ser bom ou não, mas ama o que faz... Se o cara não tem github ou não usa para nada, ele pode ser sim MUITO bom, mas não dá para saber assim de cara se o lek ama mesmo codar ou só trabalha pq precisa de dinheiro.

Na minha opinião se não for para uma demanda pontual, ou seja, se vc quer contratar alguém para realmente ser um Dev fixo e trampar com o que vier pela frente, vale MUITO MAIS um cara que ama o que faz e está aberto para criar soluções e aprender coisas novas do que um cara que é tecnicamente incrível com um tecnologia especifica. (isso se reflete bem no Github).

Para finalizar, levem em conta de que tudo o que falei é minha opinião é obviamente não se aplica a 100% dos casos, talvez nem a 50% ou talvez esteja tudo errado hahaha.

ps: Nossa comunidade, o mercado de trabalho... Tudo isso é baseado em sociedade, onde existe uma série de coisas como primeiras impressões e conclusões tiradas de aparências.. ou seja, SIM, se vc tiver no seu GH que e membro do time de alguma Lib ou projeto F#%$#, vc seria contratado por várias empresas BR sem nem fazer teste, ou só com uma entrevista p ver se vc não é um psicopata. Isso quase não acontece lá fora, onde não importa o **nome**, vc passa por mil testes e o resultado final é pela sua competência, não status.. ai o Github tem outro peso... enfim..

**Comment by fabiomirandaa** [comment]
Pelo que eu entendo, existem profissionais que amam suas profissões e não fazem apenas porque tem de trabalhar em alguma coisa. Esse parece ser o caso dos profissionais que são ativos na comunidade e consequentemente tem um github super ativo.  Dependendo da cultura da empresa, isso pode ser muito bom, pois o cara vai ajudar em projetos open-source carregando a "marca" da empresa em que trabalha, e de certa forma fazer com que a empresa seja bem vista pela comunidade.  

Porém, não é uma obrigatoriedade o cara ser um contribuidor, ter um perfil ativo no github com commits diários/semanais/mensais. Para empresa o que realmente interessa é a qualidade do desenvolvedor, e não se ele tá contribuindo pra comunidade. 

Algo que o github pode ajudar e muito é em avaliar a qualidade do código da pessoa, sem precisar submeter a um teste só pra se ter ideia disso. 

Em resumo: O Github é bom pra dar visibilidade a desenvolvedores que são ativos na comunidade, é bom pra empresas que querem ter um cultura de desenvolvimento de projetos open-source.

**Comment by malaquiasdev** [comment]
Deixando os meus 50 centavos.

Para empresa: Um GitHub ativo em um primeiro momento não quer dizer que o dev é melhor ou pior que o que não possui um GitHub. Mas server para ela analisar o perfil do profissional que ela vai contratar, porque ela pode vasculhar e observar todo historico do dev aqui. Conheço algumas empresas que a depender da necessidade de contratar ela não faz prova ou projetos, mas pede seu GitHub buscando vê algo pronto.

Para Dev: Visibilidade da sua evolução e engajamento, lembre-se que GitHub não se trata só de código, eu mesmo disponibilizo textos que público no meu blog aqui. Em algum momento da minha vida isso pode vim a ser util numa seleção de emprego para analise de perfil. Outra coisa interessante é que assim como as empresas podem vasculhar o meu historico, eu também posso usa-lo para mostrar como eu evoluir. 

Resumo: Ter um GitHub ativo não é bom somente se você esta ativo numa comunidade, é bom para que buscar melhorar e buscar oportunidades maiores. As empresas na avaliação de perfil encaram isso como um diferencial porque geralmente quem usa o GitHub costuma experimentar coisas.

**Comment by wbruno** [comment]
Quando avalio através de GitHub uso para **ver o código** do cara.
Não se ele possui algum projeto open source relevante, se abre pull requests, ou se commita todo dia.

E sim, para ver o que ele faz, como é o código dele. Assim evito até de enviar um teste ou chamar para conversar alguém que não consegue produzir um código legível, de fácil manutenção e escalável.

**Comment by diegogriep** [comment]
Como disse o @mmalaquias1 
> Deixando os meus 50 centavos.

Falando baseado nos meus 14 anos de carreira.

**O quanto é importante pra vocês, a nível de contratação, o perfil do github do desenvolvedor?**
Depende da empresa. Tem empresa que pede que você faça o código e coloque no github para avaliação, ou simplesmente, pedem para enviar por e-mail em um .zip.

**Será que estamos sendo injustos ou negligentes na hora de comparar um profissional que não tem um profile bom ou uma conta no github com outro que já é mais ativo nas comunidades?**
Você estar engajado com a comunidade open-source, acredito ser um atalho para as empresas, pois por ai, elas veem o seu código, a sua preocupação em repassar conhecimento.

**Como vocês avaliam esses profissionais (que não tem um perfil no github) quando são comparados com outros programadores? Será que o fato de ele ser ativo ou ter seus projetos públicos é um ponto forte para a contratação?**
Veja resposta anterior.

**Será que estamos deixando de contratar bons profissionais apenas pelo fato de eles não terem um perfil no github ou algo do tipo?**
Acredito que não.

**Quando é o momento certo de comparar programadores com currículos distintos (um sem e outro com github)?**
Uma boa pergunta. Será que realmente existe essa necessidade?

**E, vocês acham que ser ativo e/ou ter um profile bom no github é um requisito necessário ou desejável para uma vaga?**
Necessário, não. Mas, ajuda.

**Comment by suissa** [comment]
Um perfil bom te facilita a entrada em empresas que miram esse tipo de profissional, mais colaborativo e aberto. Muitas das vezes isso acarreta em uma performance melhor dependendo do trabalho a ser feito.

**Porém para fazer QUALQUER curso da Webschool o aluno TERÁ que usar Github, isso é o mínimo.**

**Comment by diessica** [comment]
Concordo com cada palavra do @fdaciuk, principalmente:
> [...] na hora do "vamos ver" mesmo de uma contratação, tem muitos mais pontos que pesam na decisão. https://github.com/frontendbr/forum/issues/458#issuecomment-279880101

Depois de conhecer ótimos programadores sem 1 repositório no GitHub, vejo GitHub só como um _plus_.

Teve algo que ninguém citou, e é uma das coisas que fazem o GitHub agregar bastante ao perfil profissional: o GitHub também se tornou uma rede em que devs compartilham referências e realizam discussões. Muitas pessoas, inclusive eu, usam o GitHub  para se atualizar sobre tecnologias e ferramentas – e, agora, discutir sobre elas (essa comunidade do Front-end Brasil é um exemplo).

Ou seja, ter um perfil ativo no GitHub atualmente pode significar não só que você colabora, mas também que se importa em se manter atualizado e trocar ideias sobre sua área. Um grande plus!

Mas como existem outras formas de fazer isso, isso ainda deixa o GitHub como um plus. Um grande plus, como eu disse, mas não um diferencial decisivo.

--- 

> Porém para fazer QUALQUER curso da Webschool o aluno TERÁ que usar Github, isso é o mínimo. https://github.com/frontendbr/forum/issues/458#issuecomment-280203910

Esse tópico se refere ao impacto de um perfil no GitHub no **processo de contratação**. Respeitar o tema sendo discutido no tópico é o primeiro passo para mantermos a qualidade das discussões do fórum.

**Comment by nandomoreirame** [comment]
Valeu pessoal, obrigado pela opinião de todos \o/

**Comment by malaquiasdev** [comment]
Galera Josh Wulf publicou um artigo no Medium recentemente, e fala o impactado do GitHub na carreira dele:

https://medium.com/@sitapati/the-impact-github-is-having-on-your-software-career-right-now-6ce536ec0b50#.83e6cnilg

**Comment by felipefialho** [comment]
Do carario todas as resposta. Vou deixar meus 2 centavos também.

___

Não acho que a pessoa ter perfil, ou usar ou não o Github, faz dela profissional melhor ou pior. Existe milhares e milhares de desenvolvedores de altissimo nivel, que simplesmente não curtem, não tem tempo, ou não tem cultura de participar de código aberto.

Mas como disse o @wbruno e o @fdaciuk, quando a pessoa em questão, tem um perfil ativo, fica mais fácil analisar seus códigos, suas participações e até mesmo comparar para ver se o sujeito sabe o que está fazendo ou só usa soluções prontas e altera.

Outro ponto importante, e muito importante vai de encontro ao que a @diessica disse. Eu realmente acho que os profissionais ativos no Github TENDEM (vejam só, TENDEM) a estarem mais atualizados e interessados na profissão, e claro, o resultado disso costuma ser profissionais mais completos. Por isso vejo todo sentido do mundo no grupo de Vagas daqui por exemplo, é praticamente um filtro inicial para as empresas.

**Reply by ghost** [reply]
@felipefialho Gostei do seu comentário e achei interessante, sensato.

**Comment by DevCleverton** [comment]
trabalhei muitos anos sem github!

**Comment by jeeanribeiro** [comment]
Acho que acima de tudo é importante o profissional mostrar proficiência nas habilidades necessárias pra vaga e o GitHub é um bom lugar pra isso.
Versionamento é essencial pra trabalho em equipe, então tá ai mais um motivo pra ter GitHub.
Resumindo, pra mim é o melhor jeito de provar que vc sabe o que diz que sabe
